### PR TITLE
JAVA/BUILD: Fix make dist and jar install

### PR DESCRIPTION
**Pull request by yosefe**
# Why
- Fix recent build issues on fedora30 VM
- Clean up dist tarball
- Install JAR file via generic (rather than custom) method

# How
- Make sure EXTRA_DIST includes only source files
- Install JAR by defining jardir and jardir_DATA variables

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>UNKNOWN</b> on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 09-Jan-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/bb7424d3d3cf72498fb9f3924b269099df265dcf/_main) | :question: ABORTED
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/60ab517a93c7bed86083576dbd0598dfcb73921f/hpc-arm-cavium-jenkins_W0) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/e862bcb88f2e1fa4f2c25cdbfbe7cad5567e21bd/hpc-arm-cavium-jenkins_W1) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/fce5167a54aef7dcb3c33a2f95fa620b9b420bcd/hpc-arm-cavium-jenkins_W2) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/ca8cb507ac927cd3a098edfc66585bc5740e200d/hpc-arm-cavium-jenkins_W3) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/a951bbe0a009b2fe9a25625f1ff5aff27aae7d8c/hpc-arm-hwi-jenkins_W0) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/bfc668b4f9ad1b5634a1c9e6f85b1e3cc295ca93/hpc-arm-hwi-jenkins_W1) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/9a1b649f0fe4272927d16cb84218591d0bc73204/hpc-arm-hwi-jenkins_W2) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/8d3b40ffa406678916772cc59dc237b8dd330e90/hpc-arm-hwi-jenkins_W3) | :question: UNKNOWN
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/12bb2e23b296d638dc173fe194df1164ccff3ce3/hpc-test-node-gpu_W0) | :question: UNKNOWN
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/ae3db78d308e1fd9865633376cdaa1b4f7738b2f/hpc-test-node-gpu_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/cc4fbe6134210384a9673feabf78b4f37e329834/hpc-test-node-gpu_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/0b63b1e46b8a624e2d3a39906c2de8ad6db12e91/hpc-test-node-gpu_W3) | :question: UNKNOWN
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/a2fe823933ab931551ffedc11b01f2d95e7c5755/hpc-test-node-legacy_W0) | :question: UNKNOWN
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/863ddd4b7332c25c8ea5b66f13979d1c985b7ed5/hpc-test-node-legacy_W1) | :question: UNKNOWN
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/cd6b667cac900f0322cdfac3beb59f5c0725bf2b/hpc-test-node-legacy_W2) | :question: UNKNOWN
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/b3ee813b540ad00b90191e77a3739ff42ba53caa/hpc-test-node-legacy_W3) | :question: UNKNOWN
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/6b9ee9ceb05553d03ae6b4d3f86e72f3cb38ef94/hpc-test-node-new_W0) | :question: UNKNOWN
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/b9d63d42ea65d1e8a2327d656d65644e0c246a36/hpc-test-node-new_W1) | :question: UNKNOWN
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/a8c6f1c72b11843b9aa4eb3ed818cd6bbe190ef2/hpc-test-node-new_W2) | :question: UNKNOWN
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/c0d0c73099e8909c69c4cf32640cdf31c96650c2/hpc-test-node-new_W3) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/981da076c20dde5b05dd40e3924c7303e3cd3f83/r-vmb-ppc-jenkins_W0) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/640677eef24eab65ac7c2e53a89d289694245c60/r-vmb-ppc-jenkins_W1) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/2335b1b8a648f71ff0b834baf264b8db82fc68e7/r-vmb-ppc-jenkins_W2) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/4025f6cb1fd0d28af4ae813a9c45189a/raw/66ee431d6284fcd39bdce5373a24fce8ae77e97e/r-vmb-ppc-jenkins_W3) | :question: UNKNOWN

**Comment by petro-rudenko** [comment]
Looks good, i thought to just use:
```
dist-hook:
	find $(distdir) -not -name "*.cc" -not -name "Makefile.am" -not -name "jucx_common_def.h" -mindepth 1 -delete
```
to remove all crap, but this more robust.

File: bindings/java/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	src/test \
+	checkstyle.xml \
+	pom.xml.in \
+	README.md
+	
+SUBDIRS = \
+	src/main/native
+	
+clean-local:
+	-rm -rf resources
```
Comment (petro-rudenko) [inline bindings/java/Makefile.am]: minor: new line

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 09-Jan-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/6d3febdf5b7473af061035350bd02159b7f3557d/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/354062f01c1e5d11833a31ff56bec727e7044731/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/d54dbccb3d993444dc2e3e5bb475f78a5df104fa/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/3733be54cdd11d4c1b80e53fdf4f15ad47319704/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/52ccd6f9d5554361588751a4cdb97c003c1a3520/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/b447d063b0cf33c95eb31d0cc499dd6ac47504f3/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/aee7baec86cbea468067b168eeb58bee0c201b8f/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/455991354f67246cb39edea84d47d33f417fe918/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/6424d1083b4a7150eaaa24a7464ba599c60fb604/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/7f9cb58057170294d2eb549be569b3c6374ef1b7/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/57df825674a478e43110c2892dbfb2c4b9cedbd6/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/8754daed58e680984c9755481dd9018bf1948d3b/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/af7b495083463dda5764a86ce7f2f59ea5c1257c/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/621059e770bfe06f940ef0e3c3d121985fadca30/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/301764689d8b6113dd11875012a0c0e0eb1cf350/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/20f1b61e7ccb7c92319d718860e48d0e0b1eacae/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/dd93eb1d2203869300e9ce437bf6d75731546dc5/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/9607b9d74951e29f51669b1fb5fe040cbf47c22c/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/07d138c91309a012cdc1537d978bbeb7c499b3ba/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/f35d5a4a492edc22a2a5b1d22781bccd75bd14c3/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/22de9e99ad3a20a0054e6521ddf5d798b25cf95b/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/1a3608ca0a111c1eeb982d0d4113f037462385b7/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/d99875dbdd11b7b1a979d846d0d58d2daeaadc27/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/ef7bfb094280465becf55588af25a9805707aeb5/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/3514e12db3bf68f259ffd3c3ad238633/raw/fe4d07a7f7d3ec29b6330153684abe7e5da2adb4/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**